contrib/epee/src/byte_slice.cpp

```diff
@@ -146,16 +146,18 @@ namespace epee
     : byte_slice()
   {
     std::size_t space_needed = 0;
-    for (const auto source : sources)
+    for (const auto & source : sources)
```

Comment:
Did CppCheck blindly recommend to add an `&` here? There's little benefits since its a 16-byte value with a trivial copy-constructor - the compiler is free to optimize as necessary anyway.

Reply:
No. This was my own initiative while touching the package.

The compiler optimizes this only in the release mode. I see little reason not to use a const reference, since it doesn't hurt. Especially if `auto` is used, one can't assume, that in future the `sources` remains the same 16-byte value, for which the const reference currently brings no benefit, I admit.

I can still revert it, if it's not convincing for you.